Code/Tools/FBuild/FBuildTest/Tests/TestExec.cpp

```diff
@@ -299,6 +335,45 @@ void TestExec::Build_ExecCommand_ExpectedFailures() const
     TEST_ASSERT( !fBuild.Build( targets ) );
 }
 
+//------------------------------------------------------------------------------
+void TestExec::Build_ExecEnvCommand_StdOut() const
+{
+    // Run the execenv command and ensure we get the expected output 
+    FBuildTestOptions options;
+    options.m_ConfigFile = "Tools/FBuild/FBuildTest/Data/TestExec/exec.bff";
+
+    FBuild fBuild( options );
+    fBuild.Initialize( "../tmp/Test/Exec/execenv.fdb" );
+
+    // make sure the output is where it is expected
+    const AStackString<> outFile_stdout( "../tmp/Test/Exec/Env_StdOut.txt.stdout" );
+
+    // clean up anything left over from previous runs
+    EnsureFileDoesNotExist( outFile_stdout );
+
+    // build
+    TEST_ASSERT( fBuild.Build( "ExecEnvCommandTest_StdOut" ) );
+
+    EnsureFileExists( outFile_stdout );
+
+    // Read contents
+    FileStream f;
+    f.Open( outFile_stdout.Get(), FileStream::READ_ONLY );
+    AStackString<> buffer;
+    buffer.SetLength( (uint32_t)f.GetFileSize() );
+    f.ReadBuffer( buffer.Get(), buffer.GetLength() );
+    f.Close();
+
+    // Split lines
+    Array< AString > lines;
+    buffer.Replace( '\r', '\n' );
+    buffer.Tokenize( lines, '\n' ); // Will discard empty lines
+
+    TEST_ASSERT( lines.GetSize() == 2 );
+    TEST_ASSERT( lines[0] == "FB_ENV_TEST_VALUE=FASTbuild" );
+    TEST_ASSERT( lines[1] == "PATH=" );
```

Comment:
I think the two new test functions could be collapsed together and most of the logic in this function removed if we based the test passing entirely on the return code as I mention above.